### PR TITLE
conntrack filter by protocol and port

### PR DESCRIPTION
**Pull request by aojea**
Add a new method to the ConntrackFilter to be able to
filter conntrack entries by Layer 4 protocol and ports

### CODE REVIEW

**Comment by aojea** [comment]
@aboch @vishvananda can you please take a look?

File: conntrack_linux.go [hunk 1]
```diff
@@ -354,38 +363,91 @@ func (f *ConntrackFilter) AddIP(tp ConntrackFilterType, ip net.IP) error {
...
-	}
+	match := true
 
-	// -src-nat ip    Source NAT ip
-	if elem, found := f.ipFilter[ConntrackReplySrcIP]; match && found {
-		match = match && elem.Equal(flow.Reverse.SrcIP)
-	}
+	// IP conntrack filter
+	if len(f.ipFilter) > 0 {
+
```
Comment (dcbw) [inline conntrack_linux.go:412]: @aojea added whitespace here

File: conntrack_linux.go [hunk 2]
```diff
@@ -354,38 +363,91 @@ func (f *ConntrackFilter) AddIP(tp ConntrackFilterType, ip net.IP) error {
...
+			match = match && (elem.Equal(flow.Reverse.SrcIP) || elem.Equal(flow.Reverse.DstIP))
+		}
 	}
 
-	// Match source or destination reply IP
-	if elem, found := f.ipFilter[ConntrackReplyAnyIP]; match && found {
-		match = match && (elem.Equal(flow.Reverse.SrcIP) || elem.Equal(flow.Reverse.DstIP))
+	// Layer 4 Port filter
+	if len(f.portFilter) > 0 {
+
```
Comment (dcbw) [inline conntrack_linux.go:441]: Unecessary whitespace

**Comment by dcbw** [comment]
Otherwise looks good to me

**Comment by dcbw** [comment]
LGTM!

File: conntrack_linux.go [hunk 3]
```diff
@@ -354,38 +363,89 @@ func (f *ConntrackFilter) AddIP(tp ConntrackFilterType, ip net.IP) error {
 	return nil
 }
 
+// AddPort adds a Port to the conntrack filter if the Layer 4 protocol allows it
+func (f *ConntrackFilter) AddPort(tp ConntrackFilterType, port uint16) error {
+	switch f.protoFilter {
+	// TCP, UDP, DCCP, SCTP, UDPLite
+	case 6, 17, 33, 132, 136:
+	default:
+		return errors.New("Filter attribute not available without a Layer 4 protocol")
```
Comment (aboch) [inline conntrack_linux.go:372]: Please have error msg also print the `protoFilter` value.

Comment (aboch) [reply conntrack_linux.go:372]: Also change 
`[..] without a Layer 4 protocol` => `[...] without a valid Layer 4 protocol` 
either "valid" or "supported", you choose

Comment (aojea) [reply conntrack_linux.go]: done

**Comment by aboch** [comment]
LGTM

**Pull request merged**